### PR TITLE
Fix potential null pointer dereference

### PR DESCRIPTION
**Pull request by szsam**

### CODE REVIEW

File: src/drivers/usb/hc/ohci/ohci_pci.c [hunk 1]
```diff
@@ -88,6 +88,10 @@ static void *ohci_ed_alloc(struct usb_endp *ep) {
 	struct ohci_td *sentinel_td = ohci_td_alloc();
 	struct ohci_ed *ed = pool_alloc(&ohci_eds);
 
+	if (!sentinel_td || !ed) {
+		return NULL;
```
Comment (Machiry) [inline src/drivers/usb/hc/ohci/ohci_pci.c:92]: You might have to free `sentinel_td`

Comment (szsam) [reply src/drivers/usb/hc/ohci/ohci_pci.c:96]: That's right. Fixed.

Comment (anton-bondarev) [reply src/drivers/usb/hc/ohci/ohci_pci.c:96]: I'm agree with @Machiry you have to free allocated object if one of pointers are not null.
I think it's better to check them separately


```	struct ohci_td *sentinel_td;
	struct ohci_ed *ed; = pool_alloc(&ohci_eds);
	
 	sentinel_td = ohci_td_alloc();
	if (!sentinel_td ) {
	return NULL;
	}
        ed = pool_alloc(&ohci_eds);
        if (!ed) {
       ohci_td_free(sentinel_td);
       }
       ```

**Review by anton-bondarev — APPROVED** [review]
Thanks.
LGTM

**Comment by szsam** [comment]
Oh, I haven't done rebasing. There is a fixup commit.

**Comment by anton-bondarev** [comment]
@szsam Oh, yes. It was my fault. I see now. But I think it is not so significant. Next time I will be more careful
Thank you one more time.

**Comment by anton-bondarev** [comment]
Hello @Machiry 
Thanks for the information.
I filled the survey

**Pull request merged**